drivers/gpio/gpio_sam.c

```diff
@@ -40,9 +40,15 @@ static int gpio_sam_port_configure(const struct device *dev, uint32_t mask,
 	const struct gpio_sam_config * const cfg = dev->config;
 	Pio * const pio = cfg->regs;
 
-	if (flags & GPIO_SINGLE_ENDED) {
-		/* TODO: Add support for Open Source, Open Drain mode */
+	if ((flags & GPIO_OPEN_DRAIN) == GPIO_OPEN_DRAIN) {
+		/* Enable open-drain drive mode */
+		pio->PIO_MDER = mask;
+	} else if (flags & GPIO_SINGLE_ENDED) {
```

Comment:
any reason you are not applying the same pattern as the above?

(i.e. == GPIO_SINGLE_ENDED)

Reply:
I chose not to use that pattern in the 'else if' clause because the `GPIO_SINGLE_ENDED` flag is only a single bit, so it isn't functionally necessary in the same way as the if clause, and because it was a more minimal change on the original driver (which also just checked with `flag & GPIO_SINGLE_ENDED`).

I don't have a strong opinion on this however, if it is preferable for consistencies sake to have each clause use the same pattern, I'm happy to make that change,  as it won't differ functionally; just need someone to voice the opinion that it would be preferred.

Reply:
In general, double checking the AND result is completely redundant.

You should be just doing != 0.

Also not doing != 0 is implicit integer to boolean conversion, and is against MISRA and other secure/safe coding standards.